gobblin-service/src/test/java/org/apache/gobblin/service/modules/orchestration/OrchestratorTest.java

```diff
@@ -114,6 +119,8 @@ public void setup() throws Exception {
     FlowLaunchHandler mockFlowTriggerHandler = mock(FlowLaunchHandler.class);
     DagManager mockDagManager = mock(DagManager.class);
     doNothing().when(mockDagManager).setTopologySpecMap(anyMap());
+    TopologySpecFactory mockedTopologySpecFactory = mock(TopologySpecFactory.class);
+    doReturn(Collections.singleton(this.topologySpec)).when(mockedTopologySpecFactory).getTopologies();
```

Comment:
prefer the `when().thenReturn()` form, which is typesafe

Reply:
I prefer this way, because in `when(obj.method()).thenReturn(obj)` it actually calls `obj.method()` which is rarely useful (and often generate exceptions) because obj is a mocked dummy object.

Reply:
up to you.  I do agree *with a spy* that `doReturn` is better for the reason you give, but *with a mock*, the type safety is helpful.